### PR TITLE
Add `session(venv_backend='conda')` option to use Conda environments.

### LINKED ISSUE #183: Include Conda-install package functionality
**How would this feature be useful?**
Reading the documentation , nox provides automated pip-install packages feature ( that can be defined in the configuration file).
 It would be useful to support conda-install as well , to be able to include installing miniconda/anaconda package install .

I would like to work on this as my first contribution , if anyone could help me get started, would be really helpful!

### PR DESCRIPTION
**Pull request by tswast**
* Add `session(venv_backend='conda')` option to use Conda environments.
* Add `session.conda_install()` function to install packages with
  conda, instead of pip.
* Add `conda_tests` nox session to test `nox` with dependencies from
  `conda-forge`.
* Add conda to Travis and Appveyor builds.
* Add `is_sandboxed` and `allowed_globals` to `ProcessEnv` classes.
  Remove hardcoded index path.

Towards #199

Closes #183

Follow-up to https://github.com/theacodes/nox/pull/215 in order to get Travis to pick up the PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,16 @@ matrix:
   - python: '3.7'
     env: NOXSESSION="docs"
     dist: xenial    # required for Python 3.7 (travis-ci/travis-ci#9069)
+before_install:
+  # Only test with conda on one test session.
+  - if [ ${NOXSESSION} == "tests-3.7" ] then
```
Comment (crwilcox) [inline .travis.yml]: As written this is invalid yml and I think causing the failure of travis runs.
```suggestion
  - |
     if [ ${NOXSESSION} == "tests-3.7" ] then
```

**Comment by tswast** [comment]
Started docs updates in https://github.com/theacodes/nox/pull/221

**Comment by cs01** [comment]
@tswast @crwilcox  Is anything else left outstanding? I would like to make a PR building on these changes for venv support, but am waiting for this to land.

**Comment by tswast** [comment]
@cs01 No open items that I'm aware of. Just waiting on @crwilcox or another maintainer to merge and release.

**Comment by theacodes** [comment]
This is wonderful, thanks, everyone!

**Pull request merged**